### PR TITLE
Academic year workshop email updates

### PR DESCRIPTION
**Pull request by bencodeorg**
A few updates loosely grouped under the heading of "email updates for Academic Year Workshops". They are:
- Forcing "Yes, suppress email" option to be selected when creating/updating a workshop with an AYW subject.
- Special case for AYWs to suppress post-workshop emails.
- Reenables detail changes and enrollment receipts, even if "suppress reminders" is selected.
- Change tooltips to describe what suppressing email does now.

## Testing Story

New unit tests covering behavior on the model. UI changes tested manually.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -472,17 +473,7 @@ export class WorkshopForm extends React.Component {
...
-                    <li>
-                      Will not send most email notifications to enrollees, such
-                      as enrollment receipts and workshop reminders
-                    </li>
-                    <li>
-                      Will send a post-workshop survey designed for virtual
-                      workshops
-                    </li>
-                  </ul>
+                  <p>Please update your selection if/when your plans change.</p>
```
Comment (molly-moen) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:465]: maybe just "if your plans change"?

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 2]
```diff
@@ -499,28 +490,29 @@ export class WorkshopForm extends React.Component {
...
               <ControlLabel>
-                Enable email notifications?
+                Enable workshop reminder notifications?
                 <HelpTip>
                   <p>
-                    Code.org can send email notifications about this workshop to
-                    your attendees on your behalf. Notifications may include:
+                    <strong>
+                      This functionality is disabled for all academic year
+                      workshops and virtual CSF workshops.
```
Comment (molly-moen) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:487]: should we give a reason for this here? I worry we will get a lot of questions with just a blanket "disabled" message.

File: lib/cdo/shared_constants/pd/shared_workshop_constants.rb [hunk 3]
```diff
@@ -94,6 +95,28 @@ module SharedWorkshopConstants
...
+      SUBJECT_WORKSHOP_1,
+      SUBJECT_WORKSHOP_2,
+      SUBJECT_WORKSHOP_3,
+      SUBJECT_WORKSHOP_4,
+      SUBJECT_WORKSHOP_1_2,
+      SUBJECT_WORKSHOP_3_4
+    ].freeze
+
+    # Used to suppress post-workshop emails
+    ACADEMIC_YEAR_WORKSHOP_SUBJECTS = [
```
Comment (molly-moen) [inline lib/cdo/shared_constants/pd/shared_workshop_constants.rb]: could we combine these two lists somehow? Is the reason for both that we think they may divert? For now could we set `MUST_SUPPRESS_EMAIL_SUBJECTS` to `ACADEMIC_YEAR_WORKSHOP_SUBJECTS`?

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:487]: I went back and forth with Tawny on the wording for tooltips a bunch for these, I agree maybe a little vague but trust she knows partners better than I do.

File: lib/cdo/shared_constants/pd/shared_workshop_constants.rb (hunk 3 above)
Comment (bencodeorg) [reply lib/cdo/shared_constants/pd/shared_workshop_constants.rb]: Yeah, in theory they could be different -- I'll set `MUST_SUPPRESS_EMAIL_SUBJECTS` to `ACADEMIC_YEAR_WORKSHOP_SUBJECTS`, good call.

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx (hunk 2 above)
Comment (molly-moen) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:487]: sounds good 👍

**Review by molly-moen — APPROVED** [review]

**Pull request merged**